sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/aad/webapi/AADResourceServerClientConfiguration.java

```diff
@@ -37,58 +40,87 @@
+public class AADResourceServerClientConfiguration {
 
-    private static final Logger LOGGER = LoggerFactory.getLogger(AADResourceServerOboConfiguration.class);
+    private static final Logger LOGGER = LoggerFactory.getLogger(AADResourceServerClientConfiguration.class);
 
     @Autowired
     private AADAuthenticationProperties properties;
 
     @Bean
     @ConditionalOnMissingBean({ ClientRegistrationRepository.class })
     public ClientRegistrationRepository clientRegistrationRepository() {
-        final List<ClientRegistration> oboClients = createOboClients();
+        final List<ClientRegistration> oboClients = createClients();
         if (oboClients.isEmpty()) {
-            LOGGER.warn("No client registrations are found for AAD Obo.");
+            LOGGER.warn("No client registrations are found for AAD Client.");
             return registrationId -> null;
         }
         return new InMemoryClientRegistrationRepository(oboClients);
     }
 
+    @Bean
+    @ConditionalOnMissingBean
+    OAuth2AuthorizedClientService authorizedClientService(ClientRegistrationRepository clientRegistrationRepository) {
+        return new InMemoryOAuth2AuthorizedClientService(clientRegistrationRepository);
+    }
+
     /**
-     * Use InMemoryClientRegistrationRepository to create AADOAuth2OboAuthorizedClientRepository
+     * Use InMemoryClientRegistrationRepository to create AADResourceServerOAuth2AuthorizedClientRepository
      *
      * @param repo client registration
-     * @return AADOAuth2OboAuthorizedClientRepository Bean
+     * @return AADResourceServerOAuth2AuthorizedClientRepository Bean
      */
     @Bean
     @ConditionalOnMissingBean
-    public OAuth2AuthorizedClientRepository oAuth2AuthorizedClientRepository(ClientRegistrationRepository repo) {
-        return new AADOAuth2OboAuthorizedClientRepository(repo);
+    public OAuth2AuthorizedClientRepository oAuth2AuthorizedClientRepository(
+        ClientRegistrationRepository repo,
+        OAuth2AuthorizedClientService oAuth2AuthorizedClientService) {
+        return new AADResourceServerOAuth2AuthorizedClientRepository(oAuth2AuthorizedClientService, repo);
     }
 
-    public List<ClientRegistration> createOboClients() {
+    public List<ClientRegistration> createClients() {
         List<ClientRegistration> result = new ArrayList<>();
-        for (String name : properties.getAuthorizationClients().keySet()) {
-            AuthorizationClientProperties authorizationProperties = properties.getAuthorizationClients().get(name);
-            ClientRegistration.Builder builder = createClientBuilder(name);
-            builder.scope(authorizationProperties.getScopes());
-            result.add(builder.build());
+        for (String id : properties.getAuthorizationClients().keySet()) {
+            AuthorizationClientProperties authorizationProperties = properties.getAuthorizationClients().get(id);
+            if (authorizationProperties.getAuthorizationGrantType() == null) {
```

Comment:
please add some comments here or on the method, to explain why we are using obo as defult.

Reply:
@ZhuXiaoBing-cn we need to address this comment

https://github.com/Azure/azure-sdk-for-java/pull/19808/files/a379b85bc5d789983f70318c92247999f0567983#r595838276